powerdns/utils.py

```diff
@@ -329,6 +329,14 @@ def clean_content_field(self):
             validate_soa(content)
         elif type_ in DOMAIN_NAME_RECORDS:
             validate_domain_name(content)
+            # In theory NS record can contain same name and content, and with
+            # glue record configuration it will work. However it is not good
+            # practice so we are not allowing such configuration.
+            if self.get_field('name') == content:
```

Comment:
shouldn't it be placed in separated function (`validate_name_equal_to_content` or sth like that) to follow practices above?